### ISSUE TITLE
Unsupported LANG environment breaks Rails

### ISSUE THREAD

**Issue by cpoirier** [ai/r18n#137]
If your LANG environment variable is set to a locale for which there is no translation, Rails crashes with a undefined method exception:

```
r18n-core-2.1.3/lib/r18n-core/translation.rb:86:in `merge!': undefined method `each_pair' for nil:NilClass (NoMethodError)
    from r18n-core-2.1.3/lib/r18n-core/i18n.rb:238:in `block (2 levels) in reload!'
    from r18n-core-2.1.3/lib/r18n-core/i18n.rb:231:in `each'
    from r18n-core-2.1.3/lib/r18n-core/i18n.rb:231:in `block in reload!'
    from r18n-core-2.1.3/lib/r18n-core/i18n.rb:229:in `each'
    from r18n-core-2.1.3/lib/r18n-core/i18n.rb:229:in `reload!'
    from r18n-core-2.1.3/lib/r18n-core/i18n.rb:179:in `initialize'
    from r18n-core-2.1.3/lib/r18n-core.rb:51:in `new'
    from r18n-core-2.1.3/lib/r18n-core.rb:51:in `set'
    from r18n-rails-2.1.3/lib/r18n-rails.rb:51:in `block in <top (required)>'
    from activesupport-4.2.6/lib/active_support/lazy_load_hooks.rb:38:in `instance_eval'
    from activesupport-4.2.6/lib/active_support/lazy_load_hooks.rb:38:in `execute_hook'
    from activesupport-4.2.6/lib/active_support/lazy_load_hooks.rb:45:in `block in run_load_hooks'
    from activesupport-4.2.6/lib/active_support/lazy_load_hooks.rb:44:in `each'
    from activesupport-4.2.6/lib/active_support/lazy_load_hooks.rb:44:in `run_load_hooks'
    from railties-4.2.6/lib/rails/application/finisher.rb:62:in `block in <module:Finisher>'
```

Ultimately, this error occurs because the r18n-rails-api Rails loader promises to return a Hash but doesn't handle missing translations. I presume this is unintentional.

Changing R18n::Loader::Rails.load (in r18n-rails-api/loader.rb) to use fetch() with a default solves the problem:

```
      def load(locale)
        initialize_types
        reload!
        @translations.fetch(locale.code, {})
      end
```

Additionally, at least on my Mac, the LANG environment variable uses underscores instead of dashes in the language name (`en_CA.UTF-8`, instead of `en-CA.UTF-8`). I don't think that's up to spec, so should probably be converted by the Rails bootstrap, in r18n-rails/r18n-rails.rb:

```
  env = env.split('.').first.gsub("_", "-") if env
```

And, arguably, R18n::Translation.merge!, in r18n-core/translation.rb might want to protect itself from accidental nil input:

```
    def merge!(translations, locale)
      (translations || {}).each_pair do |name, value|
```

I haven't forked the repo, so can't give you a pull request. Let me know if you need one. 

Thanks!

**Comment by ai** [comment]
Great investigation, I like to have all issues like this :).

Yes, please send a PR we need to save your name in project history :D.

**Comment by gavinkflam** [comment]
The solution is very useful, thanks @cpoirier :)

**Comment by ai** [comment]
Fixed f063d6a
